tests/openvino/tools/calibrate.py

```diff
@@ -577,10 +578,27 @@ def get_nncf_algorithms_config(compression_config):
     return nncf_algorithms
 
 
+def get_allow_reshape_input(accuracy_checker_config) -> bool:
+    for model_config in accuracy_checker_config["models"]:
+        for launcher_config in model_config["launchers"]:
+            if "allow_reshape_input" in launcher_config:
+                return launcher_config["allow_reshape_input"]
+    return False
+
+
+def maybe_reshape_model(model, accuracy_checker_config, transform_fn, dataset):
+    if not get_allow_reshape_input(accuracy_checker_config):
+        return model
+
+    inputs = transform_fn(next(iter(dataset)))
```

Comment:
Can we be sure that reshaping would be needed only once and only based on 1st sample?
I mean, it might be that model need to be reshaped from sample to sample.